spark/client/client.go

```diff
@@ -434,6 +443,151 @@ func (c *ExecutePlanClient) ToTable() (*types.StructType, arrow.Table, error) {
 	}
 }
 
+func (c *ExecutePlanClient) ToRecordBatches(ctx context.Context) (<-chan arrow.Record, <-chan error, *types.StructType) {
+	recordChan := make(chan arrow.Record, 10)
+	errorChan := make(chan error, 1)
+
+	go func() {
+		defer func() {
+			// Ensure channels are always closed to prevent goroutine leaks
+			close(recordChan)
+			close(errorChan)
```

Comment:
Does close require any error handling or is this final state.

Reply:
Final state here. Once closed, no more sends are allowed (sending on a closed channel panics), but you can still receive any buffered items, and subsequent receives yield the zero value immediately. Then attempting to close a closed channel will panic.

Reply:
this is outdated now, we use seq2 for pulls which simplifies a bit